src/main/java/com/blamejared/crafttweaker/impl/loot/conditions/LootConditionManager.java

```diff
@@ -0,0 +1,115 @@
+    }
+    
+    private static <T extends ILootConditionTypeBuilder> Field findInstanceField(final Class<T> clazz) {
+        return Arrays.stream(clazz.getDeclaredFields())
+                .filter(it -> Modifier.isStatic(it.getModifiers()))
+                .filter(it -> !Modifier.isPrivate(it.getModifiers()))
+                .filter(it -> "INSTANCE".equals(it.getName()))
+                .peek(it -> it.setAccessible(true))
+                .findFirst()
+                .orElse(null);
+    }
+    
+    private static <T extends ILootConditionTypeBuilder> Pair<Constructor<T>, Constructor<T>> findConstructors(final Class<T> clazz) {
+        /*mutable*/ Constructor<T> noArg = null;
+        /*mutable*/ Constructor<T> singleArg = null;
+        
+        for (final Constructor<?> constructor : clazz.getDeclaredConstructors()) {
+            @SuppressWarnings("unchecked") final Constructor<T> typedConstructor = (Constructor<T>) constructor;
+            
+            if (Modifier.isPrivate(typedConstructor.getModifiers())) continue;
+            
+            if (typedConstructor.getParameterCount() == 0) {
+                noArg = typedConstructor; // There cannot be more than one anyway
+            } else if (typedConstructor.getParameterCount() == 1 && typedConstructor.getParameters()[0].getType() == CTLootConditionBuilder.class) {
+                singleArg = typedConstructor;
+            }
+        }
+        
+        if (noArg != null) noArg.setAccessible(true);
+        if (singleArg != null) singleArg.setAccessible(true);
+        
+        return Pair.of(noArg, singleArg);
+    }
+    
+    @SuppressWarnings("unchecked")
+    private static <T extends ILootConditionTypeBuilder> Function<CTLootConditionBuilder, T> make(final Field instance, final Constructor<T> noArg, final Constructor<T> singleArg) {
+        if (instance != null) {
+            return ignoring(() -> LamdbaExceptionUtils.uncheck(() -> (T) instance.get(null)));
+        }
+        if (noArg != null) {
+            // This is due to type inference breaking because varargs
+            //noinspection Convert2MethodRef
+            return ignoring(() -> LamdbaExceptionUtils.uncheck(() -> noArg.newInstance()));
+        }
+        
+        return builder -> LamdbaExceptionUtils.uncheck(() -> singleArg.newInstance(builder));
+    }
+    
+    private static <T extends ILootConditionTypeBuilder> Function<CTLootConditionBuilder, T> ignoring(final Supplier<T> maker) {
+        return ignore -> maker.get();
+    }
+    
+    private static <T extends ILootConditionTypeBuilder> void register(final Pair<Class<T>, Function<CTLootConditionBuilder, T>> pair) {
+        final Class<T> typeToken = pair.getFirst();
+        final Function<CTLootConditionBuilder, T> creator = pair.getSecond();
+        if (BUILDERS.containsKey(typeToken)) {
+            throw new IllegalStateException("A builder for the given type '" + typeToken.getName() + "' was already registered");
```

Comment:
In the case this were ever hit, then the Game would crash during loading.
Is that intended?

Reply:
Definitely. This is merely added as a sanity check since it should be impossible given the fact that the registering is being done automatically and classes are singletons. If you were to crash due to this check, you would have way bigger problems than just CraftTweaker not swallowing the error.